src/types.jl

```diff
@@ -33,31 +96,66 @@ Dispatch is generally carried out using the names of the keyword arguments.
 For example, `segmentize` will only accept a `GEOS` struct with only a 
 `max_distance` keyword, and no other.
 
-It's generally a lot slower than the native Julia implementations, since
+It's generally somewhat slower than the native Julia implementations, since
 it must convert to the LibGEOS implementation and back - so be warned!
+
+## Extended help
+
+This uses the [LibGEOS.jl](https://github.com/JuliaGeometry/LibGEOS.jl) package,
+which is a Julia wrapper around the C library GEOS (https://trac.osgeo.org/geos).
 """
-struct GEOS
+struct GEOS <: CLibraryPlanarAlgorithm # SingleManifoldAlgorithm{Planar}
+    manifold::Planar
```

Comment:
do we need the manifold field if its fixed to `Planar`? Or does `Planar` have fields that don't influence the type?

Reply:
Planar doesn't but other manifolds do.  I probably need to implement an API for this rather than implicit getproperty though...

Reply:
Yeah, I would just make these fields the return value of an API function, relying on getproperty is fragile and having fixed type fields that do nothing is a bit confusing

Reply:
On the other hand, it is weird for some algorithms to store manifold but not all.  And planar is a singleton anyway so it's not like it takes extra memory or allocations...

Reply:
Your using a field for what is actually a fixed trait.

A field communicates something that can be set to a meaningful value

Reply:
Yeah I agree that we need an API function overall.  I guess it's just a bit strange to have planar special cased as a fixed return type (all other manifolds contain information and so must be stored in the algorithm, like spherical (radius) and geodesic (ellipsoid params)).

Reply:
Resolved, we added `manifold(alg)`